### DISCUSSION TITLE
Installation on Debian Sid

### DISCUSSION THREAD

**Discussion by TassieDevyl** [X3n0m0rph59/eruption#179] [Q&A]
Hi folks just wondering if anyone has successfully built or installed Eruption on Debian Sid?

**Comment by X3n0m0rph59** [comment]
Hi @TaliesinEmrys, thanks for posting this question!

Eruption should work on Debian `Sid`! To install Eruption, you can follow the [HOWTO](https://github.com/X3n0m0rph59/eruption/blob/master/docs/INSTALL.md).

The required steps are basically these:

```shell
sudo apt install git curl
sudo apt install build-essential cmake protobuf-compiler libprotobuf-dev \
  libsystemd-dev libusb-1.0-0-dev libhidapi-dev \
  libevdev-dev libudev-dev libdbus-1-dev libpulse-dev liblua5.4-dev libx11-dev \
  libxrandr-dev libgtk-3-dev libgdk-pixbuf2.0-dev libatk1.0-dev \
  libpango1.0-dev libcairo2-dev libgtksourceview-4-dev

curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
. ~/.bashrc
rustup default stable

git clone https://github.com/X3n0m0rph59/eruption
cd eruption

make && sudo make install
```

**Comment by TassieDevyl** [comment]
Thanks for that.

I followed the instructions that you provided.

Everything went smoothly until the final step, (make start)

`make start
Notifying system daemons about Eruption...
[sudo] password for dave:
Starting up Eruption daemons...
Job for eruption.service failed because the control process exited with error code.
See "systemctl status eruption.service" and "journalctl -xeu eruption.service" for details.
make: [Makefile:47: start] Error 1 (ignored)`

`systemctl status eruption.service

 eruption.service - Realtime RGB LED Driver for Linux
     Loaded: loaded (^[]8;;file://sidvicious/lib/systemd/system/eruption.service^G/lib/systemd/system/eruption.service^[]8;;^G; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Mon 2022-10-17 15:43:13 AEDT; 3min 23s ago
       Docs: ^[]8;;man:eruption(8)^Gman:eruption(8)^[]8;;^G
             ^[]8;;man:eruption.conf(5)^Gman:eruption.conf(5)^[]8;;^G
             ^[]8;;man:eruptionctl(1)^Gman:eruptionctl(1)^[]8;;^G
             ^[]8;;man:eruption-netfx(1)^Gman:eruption-netfx(1)^[]8;;^G
    Process: 667851 ExecStart=/usr/bin/eruption -c /etc/eruption/eruption.conf (code=exited, status=214/SETSCHEDULER)
    Process: 667852 ExecStopPost=/usr/bin/rm -f /run/eruption/eruption.pid ^[[0;1;38;5;185m(code=exited, status=214/SETSCHEDULER)
   Main PID: 667851 (code=exited, status=214/SETSCHEDULER)
        CPU: 1ms

Oct 17 15:43:13 sidvicious systemd[667851]: ^[[0;1;39meruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 15:43:13 sidvicious systemd[667851]: ^[[0;1;39meruption.service: Failed at step SETSCHEDULER spawning /usr/bin/eruption: Operation not permitted
Oct 17 15:43:13 sidvicious systemd[1]: Starting Realtime RGB LED Driver for Linux...
Oct 17 15:43:13 sidvicious systemd[1]: ^[[0;1;39m^[[0;1;39meruption.service: Main process exited, code=exited, status=214/SETSCHEDULER
Oct 17 15:43:13 sidvicious systemd[667852]: ^[[0;1;39meruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 15:43:13 sidvicious systemd[667852]: ^[[0;1;39meruption.service: Failed at step SETSCHEDULER spawning /usr/bin/rm: Operation not permitted
Oct 17 15:43:13 sidvicious systemd[1]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185meruption.service: Failed with result 'exit-code'.
Oct 17 15:43:13 sidvicious systemd[1]: ^[[0;1;39mFailed to start Realtime RGB LED Driver for Linux.`

Output of journalctl -xeu eruption.service

`Oct 17 12:17:38 sidvicious systemd[1]: Starting Realtime RGB LED Driver for Linux...
░░ Subject: A start job for unit eruption.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has begun execution.
░░ 
░░ The job identifier is 104.
Oct 17 12:17:38 sidvicious systemd[769]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:17:38 sidvicious systemd[769]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/eruption: Operation not permitted
░░ Subject: Process /usr/bin/eruption could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/eruption could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:17:38 sidvicious systemd[1]: eruption.service: Main process exited, code=exited, status=214/SETSCHEDULER
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit eruption.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 214.
Oct 17 12:17:38 sidvicious systemd[791]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:17:38 sidvicious systemd[791]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/rm: Operation not permitted
░░ Subject: Process /usr/bin/rm could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/rm could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:17:38 sidvicious systemd[1]: eruption.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit eruption.service has entered the 'failed' state with result 'exit-code'.
Oct 17 12:17:38 sidvicious systemd[1]: Failed to start Realtime RGB LED Driver for Linux.
░░ Subject: A start job for unit eruption.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has finished with a failure.
░░ 
░░ The job identifier is 104 and the job result is failed.
Oct 17 12:23:12 sidvicious systemd[56212]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:23:12 sidvicious systemd[56212]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/eruption: Operation not permitted
░░ Subject: Process /usr/bin/eruption could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/eruption could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:23:12 sidvicious systemd[1]: Starting Realtime RGB LED Driver for Linux...
░░ Subject: A start job for unit eruption.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has begun execution.
░░ 
░░ The job identifier is 5765.
Oct 17 12:23:12 sidvicious systemd[1]: eruption.service: Main process exited, code=exited, status=214/SETSCHEDULER
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit eruption.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 214.
Oct 17 12:23:12 sidvicious systemd[56214]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:23:12 sidvicious systemd[56214]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/rm: Operation not permitted
░░ Subject: Process /usr/bin/rm could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/rm could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:23:12 sidvicious systemd[1]: eruption.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit eruption.service has entered the 'failed' state with result 'exit-code'.
Oct 17 12:23:12 sidvicious systemd[1]: Failed to start Realtime RGB LED Driver for Linux.
░░ Subject: A start job for unit eruption.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has finished with a failure.
░░ 
░░ The job identifier is 5765 and the job result is failed.
Oct 17 12:25:20 sidvicious systemd[74272]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:25:20 sidvicious systemd[74272]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/eruption: Operation not permitted
░░ Subject: Process /usr/bin/eruption could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/eruption could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:25:20 sidvicious systemd[1]: Starting Realtime RGB LED Driver for Linux...
░░ Subject: A start job for unit eruption.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has begun execution.
░░ 
░░ The job identifier is 5888.
Oct 17 12:25:20 sidvicious systemd[1]: eruption.service: Main process exited, code=exited, status=214/SETSCHEDULER
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit eruption.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 214.
Oct 17 12:25:20 sidvicious systemd[74273]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:25:20 sidvicious systemd[74273]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/rm: Operation not permitted
░░ Subject: Process /usr/bin/rm could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/rm could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:25:20 sidvicious systemd[1]: eruption.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit eruption.service has entered the 'failed' state with result 'exit-code'.
Oct 17 12:25:20 sidvicious systemd[1]: Failed to start Realtime RGB LED Driver for Linux.
░░ Subject: A start job for unit eruption.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has finished with a failure.
░░ 
░░ The job identifier is 5888 and the job result is failed.
Oct 17 12:33:32 sidvicious systemd[145578]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:33:32 sidvicious systemd[145578]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/eruption: Operation not permitted
░░ Subject: Process /usr/bin/eruption could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/eruption could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:33:32 sidvicious systemd[1]: Starting Realtime RGB LED Driver for Linux...
░░ Subject: A start job for unit eruption.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has begun execution.
░░ 
░░ The job identifier is 6016.
Oct 17 12:33:32 sidvicious systemd[1]: eruption.service: Main process exited, code=exited, status=214/SETSCHEDULER
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit eruption.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 214.
Oct 17 12:33:32 sidvicious systemd[145579]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 12:33:32 sidvicious systemd[145579]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/rm: Operation not permitted
░░ Subject: Process /usr/bin/rm could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/rm could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 12:33:32 sidvicious systemd[1]: eruption.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit eruption.service has entered the 'failed' state with result 'exit-code'.
Oct 17 12:33:32 sidvicious systemd[1]: Failed to start Realtime RGB LED Driver for Linux.
░░ Subject: A start job for unit eruption.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has finished with a failure.
░░ 
░░ The job identifier is 6016 and the job result is failed.
Oct 17 15:41:52 sidvicious systemd[1]: Starting Realtime RGB LED Driver for Linux...
░░ Subject: A start job for unit eruption.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has begun execution.
░░ 
░░ The job identifier is 6648.
Oct 17 15:41:52 sidvicious systemd[1]: eruption.service: Main process exited, code=exited, status=214/SETSCHEDULER
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit eruption.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 214.
Oct 17 15:41:52 sidvicious systemd[1]: eruption.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit eruption.service has entered the 'failed' state with result 'exit-code'.
Oct 17 15:41:52 sidvicious systemd[1]: Failed to start Realtime RGB LED Driver for Linux.
░░ Subject: A start job for unit eruption.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has finished with a failure.
░░ 
░░ The job identifier is 6648 and the job result is failed.
Oct 17 15:43:13 sidvicious systemd[667851]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 15:43:13 sidvicious systemd[667851]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/eruption: Operation not permitted
░░ Subject: Process /usr/bin/eruption could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/eruption could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 15:43:13 sidvicious systemd[1]: Starting Realtime RGB LED Driver for Linux...
░░ Subject: A start job for unit eruption.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has begun execution.
░░ 
░░ The job identifier is 6773.
Oct 17 15:43:13 sidvicious systemd[1]: eruption.service: Main process exited, code=exited, status=214/SETSCHEDULER
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit eruption.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 214.
Oct 17 15:43:13 sidvicious systemd[667852]: eruption.service: Failed to set up CPU scheduling: Operation not permitted
Oct 17 15:43:13 sidvicious systemd[667852]: eruption.service: Failed at step SETSCHEDULER spawning /usr/bin/rm: Operation not permitted
░░ Subject: Process /usr/bin/rm could not be executed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The process /usr/bin/rm could not be executed and failed.
░░ 
░░ The error number returned by this process is ERRNO.
Oct 17 15:43:13 sidvicious systemd[1]: eruption.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit eruption.service has entered the 'failed' state with result 'exit-code'.
Oct 17 15:43:13 sidvicious systemd[1]: Failed to start Realtime RGB LED Driver for Linux.
░░ Subject: A start job for unit eruption.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit eruption.service has finished with a failure.
░░ 
░░ The job identifier is 6773 and the job result is failed.`

**Comment by X3n0m0rph59** [comment]
Hi @TaliesinEmrys I tried to reproduce the problem but did not succeed...

Eruption startup using Systemd apparently works in a `Debian Sid` VM out of the box. Did you change some configuration values in `/etc/security/limits.conf`?

The error indicates that Systemd can not set the `SCHED_FIFO` realtime scheduling class.

**Reply by TassieDevyl** [reply]
I checked `/etc/security/limits.conf` and the file hasn't been altered.
All the default entries remain commented out and no additional entries have been added.

**Reply by X3n0m0rph59** [reply]
I just double checked and tried to reproduce the problem in a Debian VM, but still no success...

Could you please try a work-around until a more correct solution is found:

```shell
sudoedit /usr/lib/systemd/system/eruption.service
```

Please find:

```systemd
CPUSchedulingPolicy=rr
CPUSchedulingPriority=20
```

And change to (comment out): 

```systemd
# CPUSchedulingPolicy=rr
# CPUSchedulingPriority=20
```

Then please save the file and then restart `eruption.service`

```shell
sudo systemctl restart eruption.service
```

**Reply by TassieDevyl** [reply]
That worked thank you.
I did as you mentioned and the service still did not start.
Then I remembered that I had not finished the final step of the install process.
cd to the directory where I git cloned then `make start`

`systemctl status eruption.service` output revealed that the service is now running and the application launches.

Thank you for your help.

**Comment by TassieDevyl** [comment]
P.S as I have a Vulcan Pro, I enabled the Experimental driver by editing `/etc/eruption/eruption.conf`, as mentioned in the documentation.

The keyboard was detected successfully and I was able to change the colour profile to that which suits my set up nicely.

I thought you would appreciate some feedback given the experimental support for the Vulcan Pro.

Thank you for all your advice and support.

**Comment by TassieDevyl** [comment]
Hi once again.
I just thought about a possibility as to why the service would not start on my Debian Sid install, as opposed to "Stock" installs of Debian that you tested.

The Distribution I used is called Siduction.

This is essentially a Debian Sid based distro with some changes to increase stability somewhat a custom kernel and a small set of additional packages which don't conform with the DSFG to facilitate greater hardware support.

The thought being that the reason for the eruption.service not starting on Siduction, may be related to the custom kernel, or one of the tweaks Siduction deploys.

Siduction is a small tight knit community team based primarily in Germany and have a good relationship with upstream Debian.

https://siduction.org/
https://forum.siduction.org/

I hope this helps.

**Comment by X3n0m0rph59** [comment]
Hi @TaliesinEmrys thanks for your reply! I finally did get around to set up a Siduction VM. The custom Eruption build, as well as the Ubuntu PPA packages all work out of the box on a default installation of Siduction. So I currently am unable to reproduce the issue...

Could you please post a dump of the installed packages on your system?

```bash
sudo dpkg-query -l
```

I have created the issue #183 to track this bug

**Comment by TassieDevyl** [comment]
Output of

`sudo dpkg-query -l`

as requested.

[packages.txt](https://github.com/X3n0m0rph59/eruption/files/9875750/packages.txt)
